### PR TITLE
Fix missing docs for diagnostic callables

### LINKED ISSUE #1822: Missing API docs for CheckZero, CheckAllZero, DumpMachine
**Describe the bug**

There are no API docs for CheckZero, CheckAllZero, DumpMachine on the website.

**To Reproduce**

* https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.diagnostics/checkzero
* https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.diagnostics/checkallzero
* https://learn.microsoft.com/en-us/qsharp/api/qsharp-lang/microsoft.quantum.diagnostics/dumpmachine

### PR DESCRIPTION
**Pull request by swernli**
Fixes #1822

### CODE REVIEW

File: library/std/src/diagnostics.qs [hunk 1]
```diff
@@ -69,11 +69,37 @@ namespace Microsoft.Quantum.Diagnostics {
         body intrinsic;
     }
 
+    /// # Summary
+    /// Checks whether a qubit is in the |0⟩ or ground state, returning true if it is.
```
Comment (DmitryVasilevsky) [inline library/std/src/diagnostics.qs:73]: "|0⟩ or ground state" gives an impression that those are different states. May be use different wording if possible? Like "AKA ground state" or something?

File: library/std/src/diagnostics.qs [hunk 2]
```diff
@@ -69,11 +69,37 @@ namespace Microsoft.Quantum.Diagnostics {
...
+    ///
+    /// # Description
+    /// This operation checks whether a qubit is in the |0⟩ or ground state. It will return true only
+    /// if the qubit is deterministically in the |0⟩ state, and will return false otherwise. This operation
+    /// does not change the state of the qubit.
```
Comment (DmitryVasilevsky) [inline library/std/src/diagnostics.qs:78]: Maybe have a remarks section that such check wouldn't be possible on an actual hardware?

**Review by DmitryVasilevsky — APPROVED** [review]

File: library/std/src/diagnostics.qs [hunk 3]
```diff
@@ -69,11 +69,45 @@ namespace Microsoft.Quantum.Diagnostics {
         body intrinsic;
     }
 
+    /// # Summary
+    /// Checks whether a qubit is in the |0⟩ state (the ground state), returning true if it is.
```
Comment (tcNickolas) [inline library/std/src/diagnostics.qs:73]: I don't think we're using "ground state" anywhere else to refer to the |0⟩ state, so I would not bring this in this doc. If somebody's writing Q#, they most likely know what |0⟩ is; and "ground state" is the kind of terminology I'd expect to see in chemistry apps, not in the basic concepts

**Review by tcNickolas — APPROVED** [review]

File: library/std/src/diagnostics.qs (hunk 3 above)
Comment (swernli) [reply library/std/src/diagnostics.qs:73]: I'll just remove it then, it's simpler that way.

File: library/std/src/diagnostics.qs (hunk 2 above)
Comment (swernli) [reply library/std/src/diagnostics.qs:78]: Added

File: library/std/src/diagnostics.qs (hunk 1 above)
Comment (swernli) [reply library/std/src/diagnostics.qs]: Removed, per Mariia's suggestion

**Pull request merged**